### ISSUE TITLE
AMD Radeon Image Sharpening - not working

### ISSUE THREAD

**Issue by diminian** [Alex313031/Thorium-Win#78]
**System Details**
 - OS: Windows 10
 - Thorium Version M117.0.5938.157 - 34th Release
 - GPU: Sapphire RX 6750 XT

**Problem**
I love this new AMD feature that comes with their graphics, which is 'Image Sharpening'.
![123](https://github.com/Alex313031/Thorium-Win/assets/8825036/8b450b86-90ac-499b-9bb7-d37f85b49803)

It usually works on desktop and all the popular applications and games by default. It does work by default in Chrome - both UI and videos are sharpened corretly.

Yet inside Thorium it doesn't seem to work correctly, even if I try to force feed Thorium to it.
Thorium UI doesn't seem to be sharpened, nor do the contents of websites.
For Youtube videos it doesn't seem to work at all too.
For Twitch videos it also doesn't work, except for a single case.
It only works in a single case for Twitch videos - in full screen view - when I toggle a Twitch video to full screen, I see it refreshed in a split-second and sharpening is applied. So it works in Twitch when the video is full screen. But once I exit full screen I see it disables immediately, in a split-second sharpened video becomes unsharpened again.

**Screenshots**
A snapshot from Chrome - Twitch:
![chrome](https://github.com/Alex313031/Thorium-Win/assets/8825036/9f15030c-2c4f-4cb7-b39a-9dab27693485)
A snapshot from Thorium - Twitch:
![thorium](https://github.com/Alex313031/Thorium-Win/assets/8825036/18215b80-462a-469c-a26b-e5f634a1d545)
The resolution of these snapshots is low, because my screen resolution is 1080p and the video is not full screen.
Pay attention to the UI of the game here or the text inside, you can easily see in Chrome sharpening is working, while it doesn't in Thorium.

**Additional Notes**
I tried playing with *chrome://flags*, tried enabling all the features that seemed remotely involved with 'encoding', 'GPU', 'acceleration' or etc., but I had no luck.
For your info, I must mention that this feature also doesn't work in Firefox. However, it does work in PaleMoon. So it can be tricky. But once again, it works in all games I played by default and I'm pretty sure the trigger to make it work is somewhere on the application side as there are no additional settings in AMD Driver Software.

Please, help!
I really like Thorium and I want to make it work. But I also grew to love image sharpening, as it makes the image crispy clear and I'm pretty sure there's a lot of people around using it, so this could overall benefit the users.
Thank you!

**Comment by gz83** [comment]
What version of Chrome are you currently using?

@diminian

**Comment by diminian** [comment]
> What version of Chrome are you currently using?
> 
> @diminian

Latest Chrome - 118.0.5993.71.
But it worked fine on earlier versions too, I've been using this feature for few months now.

**Comment by gz83** [comment]
Perhaps this feature only recognizes Chrome and only works on Chrome.

Please try other third-party browsers based on the Chromium kernel, and if they don't work either, then my reasoning is correct.

@diminian

**Comment by diminian** [comment]
Just tried Vivaldi. Works fine there.
As I mentioned, it also works fine, for example, in PaleMoon.
Moreover, it works in Thorium, when Twitch video hits full screen.

I am pretty sure this feature depends on some graphics setting inside the browser, maybe video codec or decoder which is being used is the factor, or something similar.

@gz83

**Comment by diminian** [comment]
I think I was able to fix the problem.
It indeed seems to had nothing to do with Thorium.

Here is what I did:
1) Created a new rule in AMD Driver Settings just for the Thorium browser.
2) Disabled **Surface Format Optimization** in AMD Driver Advanced Settings for Thorium specifically.
![2023-10-19 11_23_18-How to Configure Radeon™ Software to Get an Optimal Gaming Experience _ AMD - Th](https://github.com/Alex313031/Thorium-Win/assets/8825036/a5fe7888-f891-473b-8e11-76972cd71ff3)
At this point I've noticed that Thorium was suddenly sharpened. Not sure why **Disabled Surface Format Optimization** was enabled in the first place. Seems its enabled by default in AMD Driver Advanced Settings, probably happened after the driver update.
3) Toggled **Image Sharpening** OFF and back ON just for a good measure.
4) Restarted Thorium and **Image Sharpening** was now working for the browser as expected.

P.S.: I tried disabling **Surface Format Optimization** without a special rule for Thorium and for some reason it didn't suffice, and **Image Sharpening** still didn't work in such case. Only with a separate rule for Thorium it did work.

Thanks!
